resources/automate.rb

```diff
@@ -106,14 +102,11 @@
     notifies :reconfigure, 'chef_ingredient[automate]', :immediately
   end
 
-  if new_resource.enterprise.is_a?(String)
-    new_resource.enterprise = [new_resource.enterprise]
-    new_resource.enterprise.each do |ent|
-      execute "create enterprise #{ent}" do
-        command "delivery-ctl create-enterprise #{ent} --ssh-pub-key-file=/etc/delivery/builder_key.pub > /etc/delivery/#{ent}.creds"
-        not_if "delivery-ctl list-enterprises --ssh-pub-key-file=/etc/delivery/builder_key.pub | grep -w #{ent}"
-        only_if 'delivery-ctl status'
-      end
+  (new_resource.enterprise.is_a?(String) ? [new_resource.enterprise] : new_resource.enterprise).each do |ent|
```

Comment:
You can actually use `Array` here which returns a flattened array in all cases...it is also preferred to call the property directly now (e.g. now `new_resource.`):

```
Array(enterprise).each do |ent|
```

Here is an example of the `Array` method usage:
```irb
➜  irb
Welcome to IRB. You are using ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]. Have fun ;)
>> Array('foo') #=> ["foo"]
>> Array(['foo']) #=> ["foo"]
```